src/dbus_api.rs

```diff
@@ -121,7 +121,30 @@ fn list_pools(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 
 
 fn create_volumes(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+    let dbus_context = m.path.get_data();
```

Comment:
We're aiming to get the stuff having to do with actually doing the action to all happen after the stuff that involves parsing the message information. So, line 126 and 127 should be moved down to right above line 134, or thereabouts.